src/index.ts

```diff
@@ -220,4 +233,32 @@ function safeKey(key: string) {
 
 function safeProp(key: string) {
 	return /^[_$a-zA-Z][_$a-zA-Z0-9]*$/.test(key) ? `.${key}` : `[${JSON.stringify(key)}]`;
+}
+
+function stringifyString(str: string) {
+	let result = '"';
+
+	for (let i = 0; i < str.length; i += 1) {
+		const char = str.charAt(i);
+		const code = char.charCodeAt(0);
+
+		if (char === '"') {
+			result += '\\"';
+		} else if (char in escaped) {
+			result += escaped[char];
+		} else if ((code >= 0xD800 && code <= 0xDBFF)) {
+			const next = str.charCodeAt(i + 1);
+			if (next >= 0xDC00 && next <= 0xDFFF) {
+				result += char;
```

Comment:
You could append the next char here as well and increment `i`, saving a loop iteration.

Reply:
👍